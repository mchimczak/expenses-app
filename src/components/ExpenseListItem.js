import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';


const ExpenseListItem = ({ id, description, amount, createdAt }) => {

    return ( 
        <div>
            <p>{description}</p>
            <p>{ numeral(amount / 100).format('$0,0.00') }</p>
            <p>{ moment(createdAt).format("Do, MMMM, YYYY") }</p>
            <Link to={`/edit/${id}`}>Edit</Link>
        </div>
     );
};

export default ExpenseListItem;