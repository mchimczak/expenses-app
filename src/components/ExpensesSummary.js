import React from 'react';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral';
import { connect } from 'react-redux';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {

    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formatedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');

    return (
        <div>
            <p>You have {expenseCount} {expenseWord}.</p>
            <p>Total amount of your expenses: {formatedExpensesTotal}.</p>
        </div>
    )
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)

    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);