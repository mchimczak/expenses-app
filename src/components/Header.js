import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
    <header>
        <h1>Expensify</h1>
        <nav>
            <ul>
                <NavLink to="/" activeClassName="is-active" exact >Home</NavLink>
                <NavLink to="/create" activeClassName="is-active">Create</NavLink>
                <button onClick={startLogout}>Log out</button>
            </ul>
        </nav>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);