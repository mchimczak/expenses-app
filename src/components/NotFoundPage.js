import React from 'react';
import { Link } from 'react-router-dom'

const NotFoundPage = () => (
    <div>
        <Link to="/">404- Home page</Link>
    </div>
)

export default NotFoundPage