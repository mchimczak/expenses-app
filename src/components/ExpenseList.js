import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses'

export const ExpenseList = (props) => {
    console.log(props);
    let expenses;

    if (props.expenses.length === 0) {
        return expenses = <p>No list to render</p>
    } else {
        expenses = props.expenses.map(expense => {
            return (
                <ExpenseListItem key={expense.id} {...expense} />
            )
        });
    }


    return ( 
        <div>
            {expenses}
        </div>
     );
};

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);