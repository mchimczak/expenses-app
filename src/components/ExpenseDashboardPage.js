import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary'



const ExpenseDashboardPage = () => (
    <div>
        <p>New app from Dash component</p>
        <ExpensesSummary/>
        <ExpenseListFilters/>
        <ExpenseList/>
    </div>
);



export default ExpenseDashboardPage