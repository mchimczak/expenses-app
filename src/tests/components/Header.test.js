import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';
// import { startLogout } from '../../actions/auth';

test('Should render Header correctly', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header startLogout={startLogout}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Should call startLogout on button click', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header startLogout={startLogout}/>);
    wrapper.find('button').simulate('click');
    expect(startLogout).toHaveBeenCalled();
})