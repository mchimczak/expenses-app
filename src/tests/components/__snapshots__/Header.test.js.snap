// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <nav>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      to="/dashboard"
    >
      Dashboard
    </NavLink>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      to="/create"
    >
      Create
    </NavLink>
    <button
      onClick={[Function]}
    >
      Log out
    </button>
  </nav>
</header>
`;
